go_expvar/check.py

```diff
@@ -63,10 +63,38 @@ def __init__(self, name, init_config, agentConfig, instances=None):
         AgentCheck.__init__(self, name, init_config, agentConfig, instances)
         self._last_gc_count = defaultdict(int)
 
-    def _get_data(self, url):
-        r = requests.get(url, timeout=10)
-        r.raise_for_status()
-        return r.json()
+    def _get_data(self, url, instance):
+        ssl_params = {
+            'ssl': instance.get('ssl', None),
+            'ssl_keyfile': instance.get('ssl_keyfile', None),
+            'ssl_certfile': instance.get('ssl_certfile', None),
+            'ssl_verify': instance.get('ssl_verify', None),
+        }
+        for key, param in ssl_params.items():
+            if param is None:
+                del ssl_params[key]
+
+        # Load SSL configuration, if available.
+        # ssl_verify can be a bool or a string (http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification)
+        if isinstance(ssl_params.get('ssl_verify'), bool) or isinstance(ssl_params.get('ssl_verify'), str):
```

Comment:
To check if `ssl_verify` is a string it's better to use `isinstance(ssl_params.get('ssl_verify'), basestring)`, this works also when the string is a `unicode` object